static/app/views/app/index.tsx

```diff
@@ -35,201 +31,151 @@ const GlobalNotifications = HookOrDefault({
   defaultComponent: () => null,
 });
 
-function getAlertTypeForProblem(problem) {
-  switch (problem.severity) {
-    case 'critical':
-      return 'error';
-    default:
-      return 'warning';
-  }
-}
-
 type Props = {
-  api: Client;
   config: Config;
-} & RouteComponentProps<{}, {}>;
-
-type State = {
-  loading: boolean;
-  error: boolean;
-  needsUpgrade: boolean;
-  newsletterConsentPrompt: boolean;
-  user?: Config['user'];
-  organization?: Organization;
+  children: React.ReactNode;
 };
 
-class App extends Component<Props, State> {
-  state: State = {
-    loading: false,
-    error: false,
-    needsUpgrade: ConfigStore.get('user')?.isSuperuser && ConfigStore.get('needsUpgrade'),
-    newsletterConsentPrompt: ConfigStore.get('user')?.flags?.newsletter_consent_prompt,
-  };
-
-  componentDidMount() {
-    this.props.api.request('/organizations/', {
-      query: {
-        member: '1',
-      },
-      success: data => {
-        OrganizationsStore.load(data);
-        this.setState({
-          loading: false,
-        });
-      },
-      error: () => {
-        this.setState({
-          loading: false,
-          error: true,
-        });
-      },
-    });
-
-    this.props.api.request('/internal/health/', {
-      success: data => {
-        if (data && data.problems) {
-          data.problems.forEach(problem => {
-            AlertActions.addAlert({
-              id: problem.id,
-              message: problem.message,
-              type: getAlertTypeForProblem(problem),
-              url: problem.url,
-            });
-          });
-        }
-      },
-      error: () => {}, // TODO: do something?
-    });
+/**
+ * App is the root level container for all routes.
+ */
+function App({config, children}: Props) {
+  const api = useApi();
+  const {organization} = useLegacyStore(OrganizationStore);
 
-    ConfigStore.get('messages').forEach(msg => {
-      AlertActions.addAlert({
-        message: msg.message,
-        type: msg.level,
-        neverExpire: true,
-      });
-    });
+  // Command palette global-shortcut
+  useHotkeys('command+shift+p, command+k, ctrl+shift+p, ctrl+k', e => {
+    openCommandPalette();
+    e.preventDefault();
+  });
+
+  // Theme toggle global shortcut
+  useHotkeys(
+    'command+shift+l, ctrl+shift+l',
+    e => {
+      ConfigStore.set('theme', config.theme === 'light' ? 'dark' : 'light');
+      e.preventDefault();
+    },
+    [config.theme]
+  );
 
-    if (DEPLOY_PREVIEW_CONFIG) {
-      displayDeployPreviewAlert();
-    } else if (EXPERIMENTAL_SPA) {
-      displayExperimentalSpaAlert();
+  /**
+   * Loads the users organization list into the OrganizationsStore
+   */
+  async function loadOrganizations() {
+    try {
+      const data = await api.requestPromise('/organizations/', {query: {member: '1'}});
+      OrganizationsStore.load(data);
+    } catch {
+      // TODO: do something?
```

Comment:
So we used to have a `setState({error: true})` here... but we actually just never did anything with it. Ctrl+F `state` in this PR and you won't find any mention of `error`.

We probably should do something here though 🤔 maybe just log to Sentry

Reply:
I wouldn't log all the errors though. For example 401 errors are not worth sending back to sentry.

Reply:
I will follow up with this, since right now it isn't doing anything either, this comment just makes it more explicit.